src/software/thunderscope/common/fps_widget.py

```diff
@@ -0,0 +1,90 @@
+import time
+from PyQt6.QtWidgets import *
+from software.thunderscope.common.frametime_counter import FrameTimeCounter
+
+
+class FrameTimeWidget(QWidget):
+    """
+    Display the fps and frametime of thunderscope. 
+    This is measured in two different places, the buffer callback in GLWidget, and
+    the refresh function in tab.
+
+    Also, this widget update every 0.5 seconds
+    """
+
+    def __init__(
+        self,
+        buffer_counter: FrameTimeCounter,
+        refresh_counter: FrameTimeCounter,
+        update_delta=0.5,
+    ):
+        """
+        Initialize FrameTimeWidget 
+
+        :buffer_counter: a frametime counter at the GLWidget Widget
+        :refresh_counter: a frametime counter at the refresh function
+        """
+        super().__init__()
+        self.buffer_counter = buffer_counter
+        self.refresh_counter = refresh_counter
+
+        self.fps_label = QLabel("some string to be show")
+        self.fps_label.setText("some fps: ")
+        self.vertical_layout = QVBoxLayout()
+        self.vertical_layout.addWidget(self.fps_label)
+
+        self.setLayout(self.vertical_layout)
+
+        self.last_update_time = time.time()
+        self.update_delta = update_delta  # updating every 0.5 seconds
+
+    def refresh(self):
+        """
+        Updating the fps based on the frametime_counter
+        """
+        # updating every self.update_delta time, which is likely 0.5 seconds
+        if time.time() - self.last_update_time < self.update_delta:
+            return
+
+        # pull the data from frametime counter, the * 1000 is to display data ms instead of s
+        buffer_frametime = self.buffer_counter.get_last_frametime() * 1000
```

Comment:
nit: We should be using these constants for converting between units

https://github.com/UBC-Thunderbots/Software/blob/5e7209758bcf150b55349e35d286c5e650d3fec9/src/shared/constants.h#L96-L99